protected/controllers/NessusmappingController.php

```diff
@@ -378,12 +378,41 @@ public function actionFilterVulns() {
 
                 throw new Exception($errorText);
             }
+            $sortBy = null;
+            $sortDirection = null;
 
-            $vulns = NessusMappingVuln::model()->orderByPluginName()->findAllByAttributes([
-                "nessus_host" => $form->hosts,
-                "nessus_rating" => $form->ratings,
-                "nessus_mapping_id" => $form->mappingId
-            ]);
+            switch ($form->sortDirection) {
+                case NessusMapping::FILTER_SORT_ASCENDING:
+                    $sortDirection = "ASC";
+                    break;
+
+                case NessusMapping::FILTER_SORT_DESCENDING:
+                    $sortDirection = "DESC";
+                    break;
+            }
+
+            switch ($form->sortBy) {
+                case NessusMapping::FILTER_SORT_ISSUE:
+                    $sortBy = "nessus_plugin_name";
+                    break;
+
+                case NessusMapping::FILTER_SORT_RATING:
+                    $sortBy = "rating";
+                    break;
+
+                case NessusMapping::FILTER_SORT_CHECK:
+                    $sortBy = "checks.name";
+                    break;
+            }
+
+            $criteria = new CDbCriteria();
+            $criteria->addCondition("nessus_mapping_id = :mapping_id");
+            $criteria->params = ["mapping_id" => $form->mappingId];
+            $criteria->addInCondition('nessus_host', $form->hosts);
+            $criteria->addInCondition('nessus_rating', $form->ratings);
+            $criteria->join = 'LEFT JOIN checks ON checks.id = check_id';
+            $criteria->order = $sortBy." ".$sortDirection;
```

Comment:
- `$sortBy . " " . $sortDirection` -- see code style guideline
- sortBy or sortDirection may be null here, making the SQL incorrect - see form validation

Reply:
sortBy and sortDirection can not be null, if you see that error just hard reload page